Tasks/UseDotNetV2/usedotnet.ts

```diff
@@ -2,54 +2,89 @@
 import * as path from 'path';
 
 import * as tl from 'azure-pipelines-task-lib/task';
-
 import { DotNetCoreVersionFetcher } from "./versionfetcher";
+import { globalJsonFetcher } from "./globaljsonfetcher";
 import { VersionInstaller } from "./versioninstaller";
 import { Constants } from "./versionutilities";
 import { VersionInfo, VersionParts } from "./models"
 import { NuGetInstaller } from "./nugetinstaller";
+import { error } from 'util';
 
 async function run() {
+    let useGlobalJson: boolean = tl.getBoolInput('useGlobalJson');
     let packageType = tl.getInput('packageType') || "sdk";
-    let versionSpec = tl.getInput('version');
-    if (versionSpec) {
-        console.log(tl.loc("ToolToInstall", packageType, versionSpec));
-        let installationPath = tl.getInput('installationPath');
-        if (!installationPath) {
-            installationPath = path.join(tl.getVariable('Agent.ToolsDirectory'), "dotnet");
-        }
-        let includePreviewVersions: boolean = tl.getBoolInput('includePreviewVersions');
+    let versionSpec = tl.getInput('version');    
+    const nugetVersion = tl.getInput('nugetVersion') || '4.4.1';
+
+    let installationPath = tl.getInput('installationPath');
+    if (!installationPath || installationPath.length == 0) {
+        installationPath = path.join(tl.getVariable('Agent.ToolsDirectory'), "dotnet");
+    }
 
-        var versionSpecParts = new VersionParts(versionSpec);
+    let performMultiLevelLookup = tl.getBoolInput("performMultiLevelLookup", false);
+    // Check if we want install dotnet
+    if (versionSpec || (useGlobalJson && packageType == "sdk")) {                
+        let includePreviewVersions: boolean = tl.getBoolInput('includePreviewVersions');
+        let workingDirectory: string | null = tl.getPathInput("workingDirectory", false) || null;        
+        await installDotNet(installationPath, packageType, versionSpec, useGlobalJson, workingDirectory, includePreviewVersions);
+    }    
+    tl.prependPath(installationPath);    
+    // Set DOTNET_ROOT for dotnet core Apphost to find runtime since it is installed to a non well-known location.
+    tl.setVariable('DOTNET_ROOT', installationPath);
+    // By default disable Multi Level Lookup unless user wants it enabled.
+    tl.setVariable("DOTNET_MULTILEVEL_LOOKUP", !performMultiLevelLookup ? "0" : "1");    
+    // Add dot net tools path to "PATH" environment variables, so that tools can be used directly.
```

Comment:
To maintain earlier behaviour, can you move 
1. tl.prependPath(installationPath);
2. tl.setVariable('DOTNET_ROOT', installationPath);
3. tl.setVariable("DOTNET_MULTILEVEL_LOOKUP", !performMultiLevelLookup ? "0" : "1");

to inside the if (versionSpec || (UseGlobalJson && .... 

as the above 3 needs only to be done if there was a version installation

Reply:
yesterday I moved this out of the function `installDotNet`. If I but it back in the if I have the same behavior? Why I can't leave this code inside of `installDotNet`?

Reply:
The current task behaviour is to set the above mentioned environment variables only when a sdk/runtime version is installed. The changes right now is diverting from the current behaviour and setting these environment variables even when a version is not installed 

Hence, I was asking to put the above 3 inside the if condition.

The current implementation of `installDotNet` seems good as it is not concerned by anything else but purely installing the versions. Thus, we can keep `installDotNet`'s behaviour as is and achieve the same by keeping these in the caller function. Also multi level lookup doesn't really go with installation process while the other can still be included.